graph.py

```diff
@@ -0,0 +1,71 @@
+from itertools import product
+import random
+from collections import deque
```

Comment:
In general I agree with you (self-documenting code) but for common standard library modules/objects do you think it's necessary?